stdlib/LinearAlgebra/src/symmetric.jl

```diff
@@ -389,9 +407,9 @@ transpose(A::Hermitian) = Transpose(A)
 
 real(A::Symmetric{<:Real}) = A
 real(A::Hermitian{<:Real}) = A
-real(A::Symmetric) = Symmetric(real(A.data), sym_uplo(A.uplo))
-real(A::Hermitian) = Hermitian(real(A.data), sym_uplo(A.uplo))
-imag(A::Symmetric) = Symmetric(imag(A.data), sym_uplo(A.uplo))
+real(A::Symmetric) = Symmetric(applytri(parent ∘ real, A), sym_uplo(A.uplo))
```

Comment:
Do you really mean first applying `real` and only then `parent`, or the other way around?

Reply:
The full operation is `parent(real(UpperTriangular(parent(::Symmetric))))`. The `parent ∘ real` is supplied as the argument that is applied to `UpperTriangular(parent(::Symmetric))` within `applytri`.

Reply:
Right, now I see it. But then I wonder if you may wish to "hardcode" the unwrapping of the "artificial" triangular wrapper into ` applytri`? It seems to be present in any occurrence below.

Reply:
`isdiag` is the only place where that's not present. Maybe we can create a separate function that extracts the outermost `parent` so that it doesn't need to be supplied everywhere? This would likely make the code easier to read as well.

Reply:
I have added an explicit function `parentof_applytri`, which performs the composition with parent and makes the call site easier to read. Suggestions on the name of the function are welcome if this seems clunky.